### DISCUSSION TITLE
LangSmith support for LlamaIndex

### DISCUSSION THREAD

**Discussion by souravdas96** [langchain-ai/langsmith-sdk#686] [Q&A]
In their Observability section, [LlamaIndex](https://docs.llamaindex.ai/en/stable/module_guides/observability/#observability) provided various tools to debug and trace RAG applications, but LangSmith is not mentioned. I was wondering if it is possible to use LangSmith with LLamaIndex.

I saw that it is possible to use `traceable` decorators and wrappers to trace the flow of the application, but it is a bit tedious to put it everywhere in the code. As for LangChain, setting it up is pretty straightforward and it starts tracing everything out of the box. Is there something similar for LlamaIndex with LangSmith as well?

**Comment by hinthornw** [comment]
Great question! You could probably create a langsmith Llama Index tracer that would work. Under the hood, it could use the traceable decorator.

**Comment by oliveox** [comment]
use langsmith through traceloop

traceloop has support for llamaindex

https://docs.llamaindex.ai/en/stable/module_guides/observability/#openllmetry

https://docs.smith.langchain.com/observability/how_to_guides/trace_with_opentelemetry#logging-traces-with-the-traceloop-sdk
